sdk/azidentity/default_azure_credential.go

```diff
@@ -51,7 +53,10 @@ func NewDefaultAzureCredential(options *DefaultAzureCredentialOptions) (*Chained
 	}
 	// if no credentials are added to the slice of TokenCredentials then return a CredentialUnavailableError
 	if len(creds) == 0 {
-		return nil, &CredentialUnavailableError{CredentialType: "Default Token Credential", Message: errMsg}
+		err := &CredentialUnavailableError{CredentialType: "Default Azure Credential", Message: errMsg}
+		azcore.Log().Write(azcore.LogError, logCredentialError(err.CredentialType, err))
+		return nil, err
 	}
+	azcore.Log().Write(LogCredential, "Azure Identity => NewDefaultAzureCredential() invoking NewChainedTokenCredential()")
```

Comment:
Does this add any value?  Looks like there is logging that happens in `NewChainedTokenCredential()`, is that enough?

Reply:
Yes, I think we might have talked about this in V1 of the logging PR 😅. It's important so users can understand why they will get ChainedTokenCredential log messages after this point. Since DAC will call that.